tests/grid/grid_component.test.ts

```diff
@@ -1953,27 +1971,61 @@ describe("Copy paste keyboard shortcut", () => {
     setCellContent(model, "B2", "b2");
     selectCell(model, "B2");
     keyDown({ key: "D", ctrlKey: true });
-    expect(model.dispatch("COPY_PASTE_CELLS_ABOVE")).toBeSuccessfullyDispatched();
+    expect(getCell(model, "B2")?.content).toBe("b1");
 
     setCellContent(model, "C1", "c1");
     setCellContent(model, "D1", "d1");
     setSelection(model, ["B2:D2"]);
     keyDown({ key: "D", ctrlKey: true });
-    expect(model.dispatch("COPY_PASTE_CELLS_ABOVE")).toBeSuccessfullyDispatched();
+    expect(getCell(model, "B2")?.content).toBe("b1");
+    expect(getCell(model, "C2")?.content).toBe("c1");
+    expect(getCell(model, "D2")?.content).toBe("d1");
   });
 
-  test("can copy and paste cell(s) on left using CTRL+R", async () => {
+  test("banane", () => {
     setCellContent(model, "A1", "a1");
-    setCellContent(model, "B1", "b1");
-    selectCell(model, "B1");
-    keyDown({ key: "R", ctrlKey: true });
-    expect(model.dispatch("COPY_PASTE_CELLS_ON_LEFT")).toBeSuccessfullyDispatched();
+    merge(model, "A2:A3");
+    setSelection(model, ["A1:A3"]);
+    handleCopyPasteResult(env, { type: "COPY_PASTE_CELLS_ON_ZONE" });
+    // @ts-ignore
+    const notificationStore = env.__spreadsheet_stores__.get(NotificationStore);
+    expect(notificationStore.raiseError).toHaveBeenCalled();
+  });
 
+  test("can copy and paste cell(s) on left using CTRL+R", async () => {
     setCellContent(model, "A2", "a2");
+    setCellContent(model, "B2", "b2");
+    selectCell(model, "B2");
+    keyDown({ key: "R", ctrlKey: true });
+    expect(getCell(model, "B2")?.content).toBe("a2");
+
     setCellContent(model, "A3", "a3");
-    setSelection(model, ["B1:B3"]);
+    setCellContent(model, "A4", "a4");
+    setSelection(model, ["B2:B4"]);
     keyDown({ key: "R", ctrlKey: true });
-    expect(model.dispatch("COPY_PASTE_CELLS_ON_LEFT")).toBeSuccessfullyDispatched();
+    expect(getCell(model, "B2")?.content).toBe("a2");
+    expect(getCell(model, "B3")?.content).toBe("a3");
+    expect(getCell(model, "B4")?.content).toBe("a4");
+  });
+
+  test("can copy and paste cell(s) on zone using CTRL+ENTER", async () => {
+    setCellContent(model, "A1", "a1");
+    setSelection(model, ["A1:B2"]);
+    keyDown({ key: "Enter", ctrlKey: true });
+    expect(getCell(model, "A1")?.content).toBe("a1");
+    expect(getCell(model, "A2")?.content).toBe("a1");
+    expect(getCell(model, "B1")?.content).toBe("a1");
+    expect(getCell(model, "B2")?.content).toBe("a1");
+  });
+
+  test("Alt+T -> Table", async () => {
+    setSelection(model, ["A1:A5"]);
+    await keyDown({ key: "T", altKey: true });
+    expect(model.getters.getTable({ sheetId, row: 0, col: 0 })).toMatchObject({
+      range: { zone: toZone("A1:A5") },
+    });
+    const { INSERT_TABLE } = require("../../src/actions/menu_items_actions");
+    expect(INSERT_TABLE as jest.Mock).toHaveBeenCalled();
```

Comment:
I'd remove this (and the `jest.mock("../../src/actions/menu_items_actions.ts"` at top of the file), you already tested that a table was inserted.

In general avoid mocking imports unless absolutely necessary. It makes tests harder to understand, and breaks if we ever move files around. 

Testing what the result of an action is is the superior option over testing `expect(helper).toHaveBeenCalled()` anyway. Testing that the helper has been called doesn't actually test much, the helper could be broken and not do anything, and the test wouldn't notice. Just test that a table was inserted, and it's enough 🙂